### PR TITLE
[HOLD FOR DEVOPS] Add django-cacheops

### PR DESCRIPTION
**Pull request by cwisecarver**
## Purpose

Reduce queries to postgres by caching querysets in redis.

## Changes

- Installed django-cacheops
- Removed GuidMixinQuerySet annotations that were conflicting with cacheops
- Added redis to docker-compose.yml

## Side effects

- Faster things? 
- Developers angry and confused.

@icereval If you could check the docker stuff. I tried but unrelated (accidentally upgraded postgres and fell into a docker rabbit hole of failure) errors made me give up after an hour of trying.

# This does not technically require redis to be running. It will fail gracefully but guid optimizations have been removed in this PR for conflicting with caching so it will be very, very slow. I'm currently working on putting them back in. So probably don't merge this unless you're @icereval ...

### CODE REVIEW

**Comment by cwisecarver** [comment]
Lies, it passed. 
screen shot 2017-03-02 at 3 18 47 pm

**Comment by cwisecarver** [comment]
Updated all the occurrences I could find of `qs.update()`. I'll wait for the tests to tell me how `qs.invalidated_update()` works when cacheops is disabled. Hopefully it's just a pass through.

File: framework/sessions/__init__.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def before_request():
         if not util_time.throttle_period_expired(user_session.date_created, settings.OSF_SESSION_TIMEOUT):
             if user_session.data.get('auth_user_id') and 'api' not in request.url:
                 OSFUser = apps.get_model('osf.OSFUser')
-                OSFUser.objects.filter(guids___id=user_session.data['auth_user_id']).update(date_last_login=timezone.now())
+                OSFUser.objects.filter(guids___id=user_session.data['auth_user_id']).invalidated_update(date_last_login=timezone.now())
```
Comment (sloria) [inline framework/sessions/__init__.py]: Doesn't need to happen in this PR, but it might be worth overriding `update` on `BaseModel's` QuerySet so that is always calls `invalidated_update`.

File: website/static/js/project-organizer.js [hunk 2]
```diff
@@ -35,22 +35,23 @@ function _poTitleColumn(item) {
         e.stopImmediatePropagation();
     };
     var node = item.data; // Where actual data of the node is
+    var title = $osf.decodeText(node.attributes.title);
     var css = ''; // Keep for future expandability -- Remove: item.data.isSmartFolder ? 'project-smart-folder smart-folder' : '';
     var isMypreprintsCollection = tb.options.currentView().collection.data.nodeType === 'preprints';
     if (item.data.archiving) { // TODO check if this variable will be available
-        return  m('span', {'class': 'registration-archiving'}, node.attributes.title + ' [Archiving]');
+        return m('span', {'class': 'registration-archiving'}, title + ' [Archiving]');
```
Comment (sloria) [inline website/static/js/project-organizer.js]: Not quite sure how these changes made it into this PR. Could you clean out the unrelated changes?

File: requirements.txt [hunk 3]
```diff
@@ -92,4 +92,7 @@ psycopg2==2.6.2
 sqlparse==0.2.2
 psycogreen==1.0
 django-bulk-update==1.1.10
+
+django-cacheops==3.1.2
 ciso8601==1.0.3
+django-cacheops==3.1.2
```
Comment (sloria) [inline requirements.txt]: Dupe line here.
